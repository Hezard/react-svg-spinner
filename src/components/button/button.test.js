import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from './button';

afterEach(cleanup);

test('Button', () => {
  const props = {
    className: 'btn-test',
    color: 'primary',
    size: 'lg'
  };
  const { container, rerender } = render(<Button {...props}>Button</Button>);

  expect(container.firstChild).toMatchSnapshot();

  rerender(
    <Button {...props} disabled>
      Button
    </Button>
  );

  expect(container.firstChild).toHaveClass('btn--disabled');
  expect(container.firstChild).toHaveAttribute('disabled');
});
