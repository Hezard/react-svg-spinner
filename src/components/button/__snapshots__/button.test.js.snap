// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<button
  class="btn btn-test btn--lg btn--primary"
>
  Button
</button>
`;
